test/transfer_rule/constructor.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const web3 = require('../test_lib/web3.js');
+const utils = require('../test_lib/utils.js');
+const TransferRule = artifacts.require('TransferRule');
+const { AccountProvider } = require('../test_lib/utils.js');
+const tokenRulesMock = artifacts.require('TokenRulesMock');
+
+
+contract('TransferRule::constructor', async () => {
+	contract('Negative testing for input parameters:', async (accounts) => {
+		
+		it('fails when tokenrules address is null.', async () => {
+			
+			const accountProvider = new AccountProvider(accounts);
```

Comment:
Let's have accountProvider in contract level (you could check how it's done in other tests).